src/masternodes/mn_checks.h

```diff
@@ -30,44 +30,71 @@ enum CustomTxErrCodes : uint32_t {
     Fatal = uint32_t(1) << 31 // not allowed to fail
 };
 
-enum class CustomTxType : unsigned char
+enum class CustomTxType : uint8_t
 {
     None = 0,
     // masternodes:
-    CreateMasternode    = 'C',
-    ResignMasternode    = 'R',
+    CreateMasternode      = 'C',
+    ResignMasternode      = 'R',
     // custom tokens:
-    CreateToken         = 'T',
-    MintToken           = 'M',
-    UpdateToken         = 'N', // previous type, only DAT flag triggers
-    UpdateTokenAny      = 'n', // new type of token's update with any flags/fields possible
+    CreateToken           = 'T',
+    MintToken             = 'M',
+    UpdateToken           = 'N', // previous type, only DAT flag triggers
+    UpdateTokenAny        = 'n', // new type of token's update with any flags/fields possible
     // dex orders - just not to overlap in future
 //    CreateOrder         = 'O',
 //    DestroyOrder        = 'E',
 //    MatchOrders         = 'A',
     //poolpair
-    CreatePoolPair      = 'p',
-    UpdatePoolPair      = 'u',
-    PoolSwap            = 's',
-    AddPoolLiquidity    = 'l',
-    RemovePoolLiquidity = 'r',
+    CreatePoolPair        = 'p',
+    UpdatePoolPair        = 'u',
+    PoolSwap              = 's',
+    AddPoolLiquidity      = 'l',
+    RemovePoolLiquidity   = 'r',
     // accounts
-    UtxosToAccount     = 'U',
-    AccountToUtxos     = 'b',
-    AccountToAccount   = 'B',
-    AnyAccountsToAccounts  = 'a',
+    UtxosToAccount        = 'U',
+    AccountToUtxos        = 'b',
+    AccountToAccount      = 'B',
+    AnyAccountsToAccounts = 'a',
     //set governance variable
-    SetGovVariable       = 'G',
+    SetGovVariable        = 'G',
     // Auto auth TX
-    AutoAuthPrep       = 'A',
+    AutoAuthPrep          = 'A',
+    // oracles
+    AppointOracle         = 'o',
+    RemoveOracleAppoint   = 'h',
+    UpdateOracleAppoint   = 't',
+    SetOracleData         = 'y',
 };
 
-inline CustomTxType CustomTxCodeToType(unsigned char ch) {
-    constexpr const char txtypes[] = "CRTMNnpuslrUbBaGA";
-    if (memchr(txtypes, ch, sizeof(txtypes) - 1))
-        return static_cast<CustomTxType>(ch);
-    else
-        return CustomTxType::None;
+inline CustomTxType CustomTxCodeToType(uint8_t ch) {
```

Comment:
will need to keep in mind this refactoring for the other PRs

Reply:
It was done by external dev.

Reply:
@bvbfan do you suggest we keep this change or revert to how it was before?

Reply:
We should add `ohty` in `txtypes` say `CRTMNnpuslrUbBaGAohty` so refactor is better, i think.

Reply:
Yeah I think so too, it's cleaner, ok approving this